spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jProperties.java

```diff
@@ -166,6 +173,11 @@ private void configureDriverWithDefaults(DriverConfiguration driverConfiguration
 			driverConfiguration.setDriverClassName(EMBEDDED_DRIVER);
 			return;
 		}
+		if (ClassUtils.isPresent(BOLT_DRIVER, this.classLoader)) {
+			driverConfiguration.setDriverClassName(BOLT_DRIVER);
+			driverConfiguration.setURI( DEFAULT_BOLT_URI );
+			return;
+		}
```

Comment:
my assumption on this is simply that since spring boot/platform doesn't provide this driver (or the file one) that if you add it manually, you probably want to use it, I'm ok if this isn't auto behavior, that just made sense to me.